Dbld jenkinsfile

bazsi opened the pull request:
This branch adds an initial jenkinsfile from ci.syslog-ng.com to be able to maintain that in-tree
going forward. We need to wait for the dbld image caching to go in first #2858.

kira-syslogng commented:
Build SUCCESS

kira-syslogng commented:
Build SUCCESS

kira-syslogng commented:
Build SUCCESS

lbudai commented:
@bazsi: we have a new AFTJ (Another Fantastic Travis Job) that checks the presence of signed-off  (and also tries to check the format of the commit messages)... this job is failing on your PR, 
if you want to make it happy, just add a signed-off to the following commit:
https://github.com/balabit/syslog-ng/pull/2859/commits/f70ee54f1dbc9d6c3520be34427958edde387fe7

bazsi commented:
I have rebased this against current master and added the missing signed-off-by. This can probably go in without too much hassle, as it is a new file that is not yet used by anything.

Once this is in, the distro-packages kira check can be remodeled after https://ci.syslog-ng.com/jenkins/job/bazsi-dbld-pipeline/

Which can eventually turn into complete release automation.

bazsi commented:
btw, this should only go in if https://ci.syslog-ng.com/jenkins/job/bazsi-dbld-pipeline/56/console is green. Right now that job was manually started.

kira-syslogng commented:
Build SUCCESS

bazsi commented:
Could we merge this for 3.24? it would make it a _lot_ easier to progress on release automation as the jenkinsfile would be taken from the source code.

mitzkia commented on dbld/Jenkinsfile:
```diff
@@ -0,0 +1,69 @@
+            steps {
+                echo "Step: source init"
+                sh "rm -rf syslog-ng ; mkdir syslog-ng ; cd syslog-ng"
+                sh "git clone https://github.com/balabit/syslog-ng.git && cd syslog-ng && git fetch origin +refs/heads/*:refs/remotes/origin/* +refs/pull/*:refs/remotes/origin/pr/* && git checkout ${sha1}"
```
Can we use new organization URI https://github.com/syslog-ng instead of https://github.com/balabit ?

bazsi replied:
done.

mitzkia commented on dbld/Jenkinsfile:
```diff
@@ -0,0 +1,69 @@
+            sh "cd ${platform}; dbld/rules package-${platform}"
+            archiveArtifacts artifacts: "${platform}/dbld/build/${platform}/*"
+        }
+        stage("${platform} - kira") {
```
Maybe this part (kira) is not needed for now. Later we can add the "kira" stage if we already tested/verified it.

mitzkia commented on dbld/Jenkinsfile:
```diff
@@ -0,0 +1,69 @@
+def jobs = []
+
+if (!params.PLATFORMS)
+    jobs = ["centos-7", "debian-stretch", "debian-buster", "ubuntu-xenial", "ubuntu-bionic", "ubuntu-cosmic", "fedora-30"]
```
Is there a plan for syncing this platform list with dbld/rules BUILDER_IMAGES list (for now this is a duplicated)?

kira-syslogng commented:
Build SUCCESS

kira-syslogng commented:
Build SUCCESS
